v22.1/security-reference/transport-layer-security.md

```diff
@@ -0,0 +1,280 @@
+- The **Online Certificate Revocation Protocol (OCSP)** allows relying parties to check the status of certificates in real time.
+
+Each strategy has pros and cons in terms of security and operational overhead. CockroachDB does not support CRLs. 
+
+See [Revoking Certificates in CockroachDB](#revoking-certificates-in-cockroachdb).
+
+## PKI in CockroachDB
+
+In a CockroachDB cluster, each node must be able to initiate http requests to any of the others. In a normal operating mode, these requests are be TLS encrypted with mutual authentication, requiring that each node present its own certificate, signed by the same CA.
+
+In addition, SQL clients and DB Console clients must be able to reach the nodes. The client must authenticate the server with a certificate signed by a trusted CA, and the server mustu authenticate the client, either with its own certificate, or with another method, such as username/password.
+
+Therefore, the nodes must each have a [private key/public certificate pair](#key-pairs) where the public certificates have been signed by a common CA (called the **node CA** here).
+
+If the client is to use mutual authentication the client must have a private key/public certificate pair, where the public certificate is signed by a CA trusted by the nodes, i.e. the CA's public certificate must be in the nodes' trust stores.
+
+## CockroachDB's TLS support and operating modes
+
+TLS encryption and server authentication are supported in all communication between CockroachDB nodes, and from clients to nodes.
+
+Currently, mutual TLS authentication, in which the client as well as the server uses a private key/public certificate pair to authenticate itself, is not supported in {{ site.data.products.db }}. Clients must use username/password combinations.
+
+{{ site.data.products.core }} does supports TLS authentication for clients.
+
+### Default mode 
+
+By default, CockroachDB clusters require TLS. Client connection requests must made with `sslmode=on`.
+The CockroachDB CLI `cockroach sql` command, by default, is made with `sslmode=on`.
+
+Clients always require a public certificate for the CA that issued the server certificate.
+
+If the connection is mutually TLS authenticated (i.e., if the client authentication method is certificate rather than username/password combination), then a private key/public certificate pair for the client is also required.
+
+### `--accept-sql-without-tls` mode
+
+CockroachDB clusters can be started with this option in order to allow clients to opt out of TLS server authentication.
+
+Client connections can be made with `sslmode=on`, in which case TLS authentication proceeds as in default mode.
+Client connections can also be made `sslmode=off` (or by using `cockroach sql --insecure` on the CLI), in which case TLS authentication is skipped.
+
+Regardless of whether TLS authentication is performed or skipped, TLS encryption is still used for all traffic.
+
+### `--insecure` mode
```

Comment:
I don't have all the details, but "insecure" also changes other things besides not requiring TLS.

Reply:
I believe this description is accurate "If a CockroachDB cluster is started with the --insecure flag, all TLS encryption and authentication is skipped."